src/extension.ts

```diff
@@ -1154,19 +1161,27 @@ const diffKubernetes = (callback) => {
             return;
         }
 
-        kubectl.invoke(` get -o json ${kindName}`, (result, stdout, stderr) => {
-            if (result !== 0) {
+        kubectl.invoke(` get -o ${fileFormat} ${kindName}`, (result, stdout, stderr) => {
+            if (result == 1 && stderr.indexOf('NotFound') >= 0) {
+                vscode.window.showWarningMessage(`Resource ${kindName} does not exist - this will create a new resource.`, 'Create').then((choice) => {
+                    if (choice == 'Create') {
```

Comment:
Use ===

also… is the choice necessary if there's only a single option int he warning message (`create`)?

Reply:
The choice is necesssary because there's a 'Close' option automatically supplied by VS Code (which we treat as cancellation).